### ISSUE TITLE
Async load not working after installed the latest version

### ISSUE THREAD

**Issue by arooran** [valor-software/ng2-tree#43]
Having a issue with async load. it throwing an exception. 
![image](https://cloud.githubusercontent.com/assets/4504096/22780660/5a966b6a-eeb7-11e6-8de4-dd61ee1603c1.png)

Once i pass the static data in the ngOnlint its working as expected.  once i initialized the treemodel with settings that error is gone. but tree is not rendering.

`
 public treeViewOptions: Ng2TreeSettings = {
        rootIsVisible: false
    };
`

    public taskTree: TreeModel = {
        value: '',
        settings: this.treeViewOptions
    };

`
ngOnInit(): void {

        setTimeout(t => {
            this.taskTree.children = [
                {
                    value: 'Aspect-oriented programming',
                    children: [
                        { value: 'AspectJ' },
                        { value: 'AspectC++' }
                    ]
                },
                {
                    value: 'Object-oriented programming',
                    children: [
                        {
                            value: {
                                name: 'Java',
                                setName(name: string): void {
                                    this.name = name;
                                },
                                toString(): string {
                                    return this.name;
                                }
                            } as RenamableNode
                        },
                        { value: 'C++' },
                        { value: 'C#' }
                    ]
                },
                {
                    value: 'Prototype-based programming',
                    children: [
                        { value: 'JavaScript' },
                        { value: 'CoffeeScript' },
                        { value: 'TypeScript' }
                    ]
                }
            ];

        }, 300);
}

`

**Comment by rychkog** [comment]
@arooran I am on it. I will try to deliver a fix today

**Comment by rychkog** [comment]
@arooran Also

`public treeViewOptions: Ng2TreeSettings = { rootIsVisible: false };`
```
public taskTree: TreeModel = {
    value: '',
    settings: this.treeViewOptions
};
```
settings in TreeModel should be of type TreeModelSettings (https://github.com/valor-software/ng2-tree#tree-1)

And Ng2TreeSettings should be passed to the tree element
https://github.com/valor-software/ng2-tree#settings

**Comment by arooran** [comment]
@rychkog thanks for looking at it. I will change my code.

**Comment by rychkog** [comment]
@arooran I've fixed an issue with async loading. It is available in version ng2-tree@2.0.0-alpha.3.
Thanks for noticing this!
